compiler/rustc_mir/src/interpret/cast.rs

```diff
@@ -85,6 +85,28 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 }
             }
 
+            Pointer(PointerCast::NotConstFnPointer) => {
+                let src = self.read_immediate(src)?;
+                match cast_ty.kind() {
+                    ty::FnPtr(_) => {
+                        // No change to value
+                        self.write_immediate(*src, dest)?;
+                    }
```

Comment:
Please merge these two new match arms into the existing one for `UnsafeFnPointer`... seems unnecessary to duplicate that code.